contracts/external/proxies/LiquidatorProxyV1ForSoloMargin.sol

```diff
@@ -406,8 +413,65 @@ contract LiquidatorProxyV1ForSoloMargin is
         return supplyValue >= borrowValue.add(requiredMargin);
     }
 
-    // ============ Parsing Functions ============
+    // ============ Getter Functions ============
+
+    /**
+     * Gets the current total supplyValue and borrowValue for some account. Takes into account what
+     * the current index will be once updated.
+     */
+    function getCurrentAccountValues(
+        Constants memory constants,
+        Account.Info memory account
+    )
+        private
+        view
+        returns (
+            Monetary.Value memory,
+            Monetary.Value memory
+        )
+    {
+        Monetary.Value memory supplyValue;
+        Monetary.Value memory borrowValue;
+
+        for (uint256 m = 0; m < constants.markets.length; m++) {
+            Types.Par memory par = SOLO_MARGIN.getAccountPar(account, m);
+            if (par.isZero()) {
+                continue;
+            }
+            Types.Wei memory userWei = Interest.parToWei(par, constants.markets[m].index);
+            uint256 assetValue = userWei.value.mul(constants.markets[m].price.value);
+            if (userWei.sign) {
+                supplyValue.value = supplyValue.value.add(assetValue);
+            } else {
+                borrowValue.value = borrowValue.value.add(assetValue);
+            }
+        }
+
+        return (supplyValue, borrowValue);
+    }
+
+    /**
+     * Get the updated index and price for every market.
+     */
+    function getMarketsInfo()
```

Comment:
Wouldn't `SoloMargin#getMarketWithInfo` give you this info?

Reply:
Yes but it also gets a lot of extra stuff which is unnecessary. Thinking about gas here since it's used in a non-view function